pkg/miekg/miekg_test.go

```diff
@@ -1,13 +1,483 @@
+			Ttl:    3600,
+			Type:   "A",
+			Class:  "IN",
+			Name:   "example.com",
+			Answer: "192.0.2.1",
+		},
+			Answer{
+				Ttl:    3600,
+				Type:   "A",
+				Class:  "IN",
+				Name:   "example.com",
+				Answer: "192.0.2.2",
+			}},
+		Additional:  nil,
+		Authorities: nil,
+		Protocol:    "",
+		Flags:       DNSFlags{},
+	}
+	res, _, _, _ := a.DoTargetedLookup(mc, "example.com", gc.NameServers[0], true, false)
+	verifyResult(t, res.(IpResult), []string{"192.0.2.1", "192.0.2.2"}, nil)
+}
+
+func TestQuadAWithoutFlag(t *testing.T) {
+	gc, a, mc := InitTest(t)
+	mockResults["example.com"] = Result{
+		Answers: []interface{}{Answer{
+			Ttl:    3600,
+			Type:   "A",
+			Class:  "IN",
+			Name:   "example.com",
+			Answer: "192.0.2.1",
+		},
+			Answer{
+				Ttl:    3600,
+				Type:   "AAAA",
+				Class:  "IN",
+				Name:   "example.com",
+				Answer: "2001:db8::1",
+			}},
+		Additional:  nil,
+		Authorities: nil,
+		Protocol:    "",
+		Flags:       DNSFlags{},
+	}
+
+	res, _, _, _ := a.DoTargetedLookup(mc, "example.com", gc.NameServers[0], true, false)
+	verifyResult(t, res.(IpResult), []string{"192.0.2.1"}, nil)
+}
+
+func TestOnlyQuadA(t *testing.T) {
+	gc, a, mc := InitTest(t)
+	mockResults["example.com"] = Result{
+		Answers: []interface{}{Answer{
+			Ttl:    3600,
+			Type:   "AAAA",
+			Class:  "IN",
+			Name:   "example.com",
+			Answer: "2001:db8::1",
+		}},
+		Additional:  nil,
+		Authorities: nil,
+		Protocol:    "",
+		Flags:       DNSFlags{},
+	}
+
+	res, _, _, _ := a.DoTargetedLookup(mc, "example.com", gc.NameServers[0], false, true)
+	verifyResult(t, res.(IpResult), nil, []string{"2001:db8::1"})
+}
+
+func TestAandQuadA(t *testing.T) {
+	gc, a, mc := InitTest(t)
+	mockResults["example.com"] = Result{
+		Answers: []interface{}{Answer{
+			Ttl:    3600,
+			Type:   "A",
+			Class:  "IN",
+			Name:   "example.com",
+			Answer: "192.0.2.1",
+		},
+			Answer{
+				Ttl:    3600,
+				Type:   "AAAA",
+				Class:  "IN",
+				Name:   "example.com",
+				Answer: "2001:db8::1",
+			}},
+		Additional:  nil,
+		Authorities: nil,
+		Protocol:    "",
+		Flags:       DNSFlags{},
+	}
+	res, _, _, _ := a.DoTargetedLookup(mc, "example.com", gc.NameServers[0], true, true)
+	verifyResult(t, res.(IpResult), []string{"192.0.2.1"}, []string{"2001:db8::1"})
+}
+
+func TestTwoQuadA(t *testing.T) {
+	gc, a, mc := InitTest(t)
+	mockResults["example.com"] = Result{
+		Answers: []interface{}{Answer{
+			Ttl:    3600,
+			Type:   "AAAA",
+			Class:  "IN",
+			Name:   "example.com",
+			Answer: "2001:db8::1",
+		},
+			Answer{
+				Ttl:    3600,
+				Type:   "AAAA",
+				Class:  "IN",
+				Name:   "example.com",
+				Answer: "2001:db8::2",
+			}},
+		Additional:  nil,
+		Authorities: nil,
+		Protocol:    "",
+		Flags:       DNSFlags{},
+	}
+	res, _, _, _ := a.DoTargetedLookup(mc, "example.com", gc.NameServers[0], false, true)
+	verifyResult(t, res.(IpResult), nil, []string{"2001:db8::1", "2001:db8::2"})
+}
+
+func TestNoResults(t *testing.T) {
```

Comment:
What does it mean for there to be no answers and the status to be NOERROR? Is this the empty non terminal case?

Reply:
Updated to return empty result in case of `NOERROR`